desktop/src/views/Pro/CreateWorkspace/CreateWorkspace.tsx

```diff
@@ -34,6 +36,31 @@ export function CreateWorkspace() {
   const { host, currentProject, managementSelfQuery, client } = useProContext()
   const navigate = useNavigate()
 
+  const { data: templates, isLoading: isTemplatesLoading } = useTemplates()
+
+  const presets = templates?.presets
+
+  const [presetId, setPresetId] = useState<string | undefined>(undefined)
+  const routerLocation = useLocation()
+
+  useEffect(() => {
+    const searchParams = new URLSearchParams(routerLocation.search)
+    const fromPreset = searchParams.get("fromPreset")
+
+    if (fromPreset && !presetId) {
+      setPresetId(fromPreset)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  const preset = useMemo(() => {
+    if (!presetId) {
+      return undefined
+    }
+
+    return (presets ?? []).find((p) => p.metadata?.name === presetId)
+  }, [presetId, presets])
```

Comment:
Couldn't all of this just go into the `useMemo`?

Reply:
No, I want the router query only to be evaluated on the first render, it will not get updated when I change the preset while in the view. The details of which preset we have, however, can change.